### PR TITLE
feat(ts): expose built files in umd

### PR DESCRIPTION
**Pull request by Haroenv**
**Summary**

Exposes the entry point for umd for each of the umd entry points.

This is achieved using the dtsRollup feature of api-extractor, which isn't needed for the es build, as local files work perfectly fine in that case.

Make sure to check the generated files when reviewing this PR

**Result**

A situation like https://github.com/algolia/create-instantsearch-app/pull/464 is done easier by simply declaring what's import from the umd

### CODE REVIEW

**Comment by Haroenv** [comment]
https://codesandbox.io/s/instantsearch-js-forked-34p6s?file=/src/app.js proves that it works 🙌

File: scripts/typescript/extract.js [hunk 1]
```diff
@@ -47,3 +41,24 @@ if (!result.succeeded) {
...
+  path.join(__dirname, '../../dist/instantsearch.production.d.ts'),
+  path.join(__dirname, '../../dist/instantsearch.production.min.d.ts')
+);
+shell.cp(
+  path.join(__dirname, '../../dist/instantsearch.production.d.ts'),
+  path.join(__dirname, '../../dist/instantsearch.development.d.ts')
+);
+shell.cp(
+  path.join(__dirname, '../../dist/instantsearch.production.d.ts'),
+  path.join(__dirname, '../../dist/instantsearch.development.min.d.ts')
```
Comment (eunjae-lee) [inline scripts/typescript/extract.js:63]: in which case do people need development.d.ts file?

Comment (Haroenv) [reply scripts/typescript/extract.js:63]: all the files are the same, but typescript works by matching a .d.ts file to a .js file, so if you're importing /dist/development.min.js then the .d.ts is needed for that to work

Comment (eunjae-lee) [reply scripts/typescript/extract.js:63]: Thanks for explaining!

**Review by eunjae-lee — APPROVED** [review]

**Review by francoischalifour — APPROVED** [review]

**Pull request merged**